### DISCUSSION TITLE
How to save data to file?

### DISCUSSION THREAD

**Discussion by ghost** [pyinstaller/pyinstaller#5825] [Help]
Currently, I build application for save data to json file,  after click button Submit. 

![image](https://user-images.githubusercontent.com/21364546/117938814-a18bdf80-b331-11eb-986d-b5d7a9f71339.png)

After, I received error related permission
`[Errno 30] Read-only file system: '/private/var/folders/__/45s3pt8n25n710kbqnvn36r299000gn/T/AppTranslocation/3343AACA-0A49-472C-A10C-0543EBD95664/d/TEST.app/Contents/MacOS/public/data.json'`

I am using command line below for packaging
`pyinstaller ui/windows.py  --add-data public:public --windowed  --name "TEST" --icon public/hiddenfolder.icns  --additional-hooks-dir=ui/.
`
Error on macos 10.13 or later. But it's work on MacOS 10.12
I'm using :
`Pyinstaller==4.3
PyQt5==5.10`

**Comment by rokm** [comment]
Look up "App Translocation" on macOS, and decide how to deal with it (on the other hand, does data.json really need to be saved into app bundle?)
